antarest/study/storage/variantstudy/model/command/create_xpansion_matrix.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2025, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+from typing import Any, Dict, List, Optional, Tuple
+
+from pydantic import Field, ValidationInfo, field_validator
+from typing_extensions import override
+
+from antarest.core.utils.utils import assert_this
+from antarest.matrixstore.model import MatrixData
+from antarest.study.business.model.xpansion_model import XpansionResourceFileType
+from antarest.study.storage.rawstudy.model.filesystem.config.model import FileStudyTreeConfig
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.variantstudy.business.utils import strip_matrix_protocol, validate_matrix
+from antarest.study.storage.variantstudy.model.command.common import CommandName, CommandOutput
+from antarest.study.storage.variantstudy.model.command.icommand import ICommand, OutputTuple
+from antarest.study.storage.variantstudy.model.command.xpansion_common import (
+    apply_config_create_resource_commands,
+    apply_create_resource_commands,
+)
+from antarest.study.storage.variantstudy.model.command_listener.command_listener import ICommandListener
+from antarest.study.storage.variantstudy.model.model import CommandDTO
+
+
+class AbstractCreateXpansionMatrix(ICommand):
+    # Command parameters
+    # ==================
+
+    filename: str
+    matrix: List[List[MatrixData]] | str = Field(validate_default=True)
```

Comment:
@MartinBelthle @TheoPascoli an after the fact comment:
we should strive to not have `List[List[float]]` in memory, it's really heavy in memory and slow to process, we should use numpy arrays instead anytime we can